pxr/base/arch/fileSystem.cpp

```diff
@@ -532,24 +532,34 @@ ArchGetFileName(FILE *file)
     }
     return result;
 #elif defined (ARCH_OS_WINDOWS)
-    static constexpr DWORD bufSize =
-        sizeof(FILE_NAME_INFO) + sizeof(WCHAR) * 4096;
-    HANDLE hfile = _FileToWinHANDLE(file);
-    auto fileNameInfo = reinterpret_cast<PFILE_NAME_INFO>(malloc(bufSize));
     string result;
-    if (GetFileInformationByHandleEx(
-            hfile, FileNameInfo, static_cast<void *>(fileNameInfo), bufSize)) {
+    WCHAR filePath[MAX_PATH];
+    HANDLE hfile = _FileToWinHANDLE(file);
+    if (GetFinalPathNameByHandleW(hfile, filePath, MAX_PATH, VOLUME_NAME_DOS)) {
         size_t outSize = WideCharToMultiByte(
-            CP_UTF8, 0, fileNameInfo->FileName,
-            fileNameInfo->FileNameLength/sizeof(WCHAR),
+            CP_UTF8, 0, filePath,
+            wcslen(filePath),
             NULL, 0, NULL, NULL);
         result.resize(outSize);
         WideCharToMultiByte(
-            CP_UTF8, 0, fileNameInfo->FileName,
-            fileNameInfo->FileNameLength/sizeof(WCHAR),
+            CP_UTF8, 0, filePath,
+            -1,
             &result.front(), outSize, NULL, NULL);
+
+        if (result.length() > 4)
+        {
+            // It needs to strip the path prefix as
+            // the path returned is DOS device path, and the
+            // syntax is one of:
+            //   \\.\C:\Test\Foo.txt
+            //   \\?\C:\Test\Foo.txt
+            if (result.compare(0, 4, "\\\\?\\") == 0 ||
+                result.compare(0, 4, "\\\\.\\") == 0)
+            {
+                result.erase(0, 4);
+            }
```

Comment:
While doing some research, I saw that network paths on Windows will look like "\\?\UNC\...". I think we've run into issues with network file paths in the past, so we need to figure out whether we need to deal with this specially.

from @sunyab

Reply:
I treated it specially without stripping the prefix if it's UNC path.